.github/coding_guidelines/CODING_GUIDELINES.md

```diff
@@ -0,0 +1,76 @@
+# MetaMask Coding Guidelines
+
+### 1. New Code Should be Typescript
+- New components and utilities should be written in Typescript and enforce typing.
+- Existing code should be refactored into Typescript where time allows. If you are replacing a component use Typescript.
+
+### 2. Using Functional Components and Hooks Instead of Classes
+- Use functional components and hooks as they result in more concise and readable code compared to classes.
+
+### 3. Organize Files Related to the Same Component in One Folder
+- An example of a component file structure:
+
+```.tsx
+AvatarAccount
+├── AvatarAccount.constants.ts
+├── AvatarAccount.stories.tsx
+├── AvatarAccount.styles.ts
+├── AvatarAccount.test.tsx
+├── AvatarAccount.tsx
+├── AvatarAccount.types.ts
+├── README.md
+├── __snapshots__
+│   └── AvatarAccount.test.tsx.snap
+└── index.ts
+```
+
+### 4. Follow Naming Conventions
+- You should always use PascalCase when naming components to differentiate them from other non-component TSX files. For example: *TextField*, *NavMenu*, and *SuccessButton*.
+- Use camelCase for functions declared inside components like *handleInput()* or *showElement()*.
+- When creating hooks use *withHookName()*.
+
+### 5. Avoid Repetitive Code
+- If you notice you are writing duplicated code or components, convert it into a component, utility functions or hooks that can be reused. Do this with [scalable intention](https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction).
+
+### 6. Component Optimization
+- For functional components, instead of having large return statements, break the component down into smaller sub-components.
```

Comment:
```suggestion
- For functional components, instead of having large return statements, breaking the component down into smaller sub-components.
```

Reply:
This sentence is in the imperative mood, giving a direct instruction or command to the reader, which is why "break" is preferred to "breaking".